### PR TITLE
jasmine spies

### PR DESCRIPTION
**Pull request by Sbuda32**
@mudimuteba and @decentguy Please review this topic about jasmine spies.

### CODE REVIEW

**Comment by Tsehla** [comment]
1) 
[ basically everything self-contained that acts like a black box to the outside world.] 
-> by outside world, you mean anything outside the program/code or do you refer to anything that is Ajax connection related. Pehaps you can add (.ie) to help new reader with quick clarification

2) [  - or even none at all, - or ]
-> do you meant to have those minus signs show like that?

3) [ McCoy ] who is maccoy, lol took me a while to get it, I would suggest [ as legit/real McCoy ] to cover learners like me.

4) [ let _age = 0,] underscore before variable is taken as if you have (underscoreJs), its fine in this case, but the dev community highly goes against it unless u got underscorejs as part of your code

5) [ describe or it block in ], perhaps you could like ('describe' or it 'block' in ), it helps

File: content/topics/jasmine-spies/_index.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
+---
+title: Jasmine Spies
+ready: True
+---
+
+### Spying on JavaScirpt Methods using Jasmine
+
+Jasmine is made for unit testing. Unit tests are supposed to test only one component of your application. A component can be a function, an object, a module, basically everything self-contained that acts like a black box to the outside world. You usually want to avoid that your unit tests fail because another component failed. That’s why you want to test your components in isolation as much as possible.
+
+For example, you may not want to send data to another server in your unit tests or you don’t want to manipulate a page’s DOM. But you want to make sure that the components which are responsible for these tasks get called correctly. How do we do this?
```
Comment (SweetBeard30) [inline content/topics/jasmine-spies/_index.md]: I assume they already know what jasmine is or what it does, just post a link to the jasmine topic and live out the description.

Comment (Sbuda32) [reply content/topics/jasmine-spies/_index.md]: Cool, but i just wanted to make it clear that components should be tested in isolation, i just feel like that has to be explained some how. would it be ok, if i rephrase the paragraph? @SweetBeard30

Comment (SweetBeard30) [reply content/topics/jasmine-spies/_index.md]: It's a topic on advanced Jasmine usage,  it won't be novices reading this and they should already been using Jasmine if they've reached this point.

Comment (SweetBeard30) [reply content/topics/jasmine-spies/_index.md]: Jasmine spies is an advace topic, anyone who reaches this point has probably been using jasmine already and are not new to it.

Comment (Sbuda32) [reply content/topics/jasmine-spies/_index.md]: @SweetBeard30 so i must rephrase the paragraph. is that what you are suggesting?

Comment (SweetBeard30) [reply content/topics/jasmine-spies/_index.md]: @Sbuda32 remove it.

Comment (Tsehla) [reply content/topics/jasmine-spies/_index.md]: Eish, sorry apparently I left a comment not a review. lol. https://github.com/Umuzi-org/tech-department/pull/323#pullrequestreview-323069927

**Pull request merged**